arcgis-ios-sdk-samples/Geometry/Nearest vertex/NearestVertexViewController.swift

```diff
@@ -0,0 +1,117 @@
+    lazy var nearestVertexGraphic = AGSGraphic(geometry: nil, symbol: nearestVertexSymbol)
+    
+    /// The map view managed by the view controller.
+    @IBOutlet weak var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap()
+            mapView.graphicsOverlays.add(graphicsOverlay)
+            mapView.setViewpointCenter(AGSPoint(x: -4487263.495911, y: 3699176.480377, spatialReference: .webMercator()), scale: 1e8)
+            mapView.touchDelegate = self
+        }
+    }
+    
+    /// Creates a map.
+    ///
+    /// - Returns: A new `AGSMap` object.
+    func makeMap() -> AGSMap {
+        let map = AGSMap(basemap: .topographic())
+        return map
+    }
+    
+    /// Add the graphics to the graphics overlay.
+    func addGraphicsToOverlay() {
+        graphicsOverlay.graphics.add(polygonGraphic)
+        graphicsOverlay.graphics.add(nearestCoordinateGraphic)
+        graphicsOverlay.graphics.add(tappedLocationGraphic)
+        graphicsOverlay.graphics.add(nearestVertexGraphic)
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        addGraphicsToOverlay()
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["NearestVertexViewController"]
+    }
+}
+
+extension NearestVertexViewController: AGSGeoViewTouchDelegate {
+    // MARK: - AGSGeoViewTouchDelegate
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        if mapView.callout.isHidden {
+            // If the callout is not shown, show the callout with the coordinates of the tapped location.
+            if let nearestVertexResult = AGSGeometryEngine.nearestVertex(in: polygonGraphic.geometry!, to: mapPoint),
+                let nearestCoordinateResult = AGSGeometryEngine.nearestCoordinate(in: polygonGraphic.geometry!, to: mapPoint) {
+                // Set the geometry for the tapped point, nearest coordinate point and nearest vertex point.
+                tappedLocationGraphic.geometry = mapPoint
+                nearestVertexGraphic.geometry = nearestVertexResult.point
+                nearestCoordinateGraphic.geometry = nearestCoordinateResult.point
+                // Get the distance in km to the nearest vertex in the polygon.
+                let distanceVertex = nearestVertexResult.distance / 1000
+                // Get the distance in km to the nearest coordinate in the polygon.
+                let distanceCoordinate = nearestCoordinateResult.distance / 1000
+                // Display the results on a callout of the tapped point.
+                mapView.callout.title = "Proximity result"
+                mapView.callout.detail = String(format: "Vertex dist: %.2f km. Point dist: %.2f km", distanceVertex, distanceCoordinate)
```

Comment:
Please use a measurement formatter so that:

- You don't need to do the measurement conversions yourself, and
- Measurements are automatically displayed in the most appropriate units for the user.

To do that, add a measurement formatter property to the view controller:

```swift
let distanceFormatter = MeasurementFormatter()
```

Then create measurements for each of the distances:

```swift
let distanceVertex = Measurement(value: nearestVertexResult.distance, unit: UnitLength.meters)
let distanceCoordinate = Measurement(value: nearestCoordinateResult.distance, unit: UnitLength.meters)
```

Then use the measurement formatter to create a string representation of the measurements and display that to the user:

```swift
mapView.callout.detail = String(format: "Vertex dist: %@; Point dist: %@", distanceFormatter.string(from: distanceVertex), distanceFormatter.string(from: distanceCoordinate))
```